### ISSUE TITLE
2.33.0: Missing @Override false positive for record constructor

### ISSUE THREAD

**Issue by nlarsson** [google/error-prone#4622]
When using constructor in record error-prone now gives a warning bout missing an override:
```java
public record Example(int a) {
  public Example() {
    this(1);
  }
}
``` 
```
Example.java:[4,10] [MissingOverride] <init> is an explicitly declared accessor method for a record component; expected @Override
    (see https://errorprone.info/bugpattern/MissingOverride)
  Did you mean '@Override public Example() {'?
```

Suspecting that https://github.com/google/error-prone/pull/4585 introduced this issue.

**Comment by tbroyer** [comment]
Fixed already in https://github.com/google/error-prone/commit/81ae61d2d103c5cef1bf6d29e92c2e4152a61e3c but not released yet

**Comment by nlarsson** [comment]
> Fixed already in [81ae61d](https://github.com/google/error-prone/commit/81ae61d2d103c5cef1bf6d29e92c2e4152a61e3c) but not released yet

Cool, thank you!
